### ISSUE TITLE
Vulkan 1.4 feature requirements are inconsistent

### ISSUE THREAD

**Issue by kusma** [KhronosGroup/Vulkan-Docs#2470]
Section [46.1. Feature Requirements](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#features-requirements) of the Vulkan 1.4 spec seems inconsistent with feature requirements:

> - If Vulkan 1.4 is supported, the following features must be supported:
>   - [...]
>   - [shaderInt16](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#features-shaderInt16)
>   - [...]
>   - [shaderInt8](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#features-shaderInt8)
>   - [...]
> - If [VK_KHR_shader_float16_int8](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VK_KHR_shader_float16_int8) is supported, [shaderFloat16](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#features-shaderFloat16), or [shaderInt8](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#features-shaderInt8) must be supported

Which one is it? Are both of `shaderInt16` and `shaderInt8` *always* required for Vulkan 1.4, or is only *either one of them* required if `VK_KHR_shader_float16_int8` is supported?

Similarly, the `VkPhysicalDeviceVulkan14Features`-struct has flags for e.g `shaderSubgroupRotate` and `shaderFloatControls2`, but these features are listed as required above. What's the point of allowing to query these, if the feature *has* to be supported? After all, if the application has a `VkPhysicalDeviceVulkan14Features`-struct, according to the lists of required features, these always needs to be `VK_TRUE`. So why do they even exist, seems like waste... These are just examples, there's more.

Generally speaking, I get the feeling that something went wrong when merging this into the main spec, and that not all of those features are supposed to be unconditionally required, but that's just a guess.

Could this please be clarified?

**Comment by kusma** [comment]
> These statements do not contradict - they are independent and can be separately and simultaneously true. shaderInt16 and shaderInt8 are required if Vulkan 1.4 is supported. The later statement about VK_KHR_shader_float16_int8 doesn't change that.

They do actually somewhat contradict. Both of these are from the same list of feature requirements for Vulkan 1.4. The wording of the `VK_KHR_shader_float16_int8`-part suggests that `shaderInt8` might be optional, but that's not the case.

> Also of note is that shaderFloat16 and shaderInt16 are different features - it seems like you've confused the two?

Yeah, I did indeed confuse them briefly, but that doesn't take away from the unclear spec wording here, I think.

> Features need to be passed into device creation as well where they can be enabled or disabled independently, which has a meaningful impact on some implementations for a number of features, which is the primary reason they are there. 

Ah yes, that's a good point. Thanks for reminding me of that.

> Based on what you've said, everything is as intended?

I still think the `VK_KHR_shader_float16_int8`-part is unclear, as I suggested above. Perhaps it's not factually incorrect, but it's not what I consider clear wording.

**Comment by Tobski** [comment]
> They do actually somewhat contradict. Both of these are from the same list of feature requirements for Vulkan 1.4. The wording of the VK_KHR_shader_float16_int8-part suggests that shaderInt8 might be optional, but that's not the case.

I'm confused by this assertion - the 1.4 bullet is independent from the VK_KHR_shader_float16_int8 bullet; the latter has nothing to do with Vulkan 1.4's requirements. They're fully independent requirements based on different things being advertised by the driver, unless you're seeing something I'm not?

> I still think the VK_KHR_shader_float16_int8-part is unclear, as I suggested above. Perhaps it's not factually incorrect, but it's not what I consider clear wording.

I won't disagree with that, we are aiming to eventually move to the feature requirements just being part of the structures that define the features, and a list in the appendix for each extension/version, rather than a single giant list which can be a bit hard to parse.

**Comment by kusma** [comment]
Oh, you're right. My bad, this is actually very clear and I can't read. Sorry for the noise.
